### DISCUSSION TITLE
Episode 135: 🇸🇮 Slovenia 🇸🇮 2023 Road Trip!

### DISCUSSION THREAD

**Discussion by codereport** [codereport/adsp2#26] [Episode]
Episode Link: https://adspthepodcast.com/2023/06/23/Episode-135.html

Feel free to write comments or ask questions :slightly_smiling_face:

**Comment by roibarkan** [comment]
On the topic of algorithms/ranges that are inherently monotonic, I'd like to mention that iota and cartesian_product produce such ranges, and that there are algorithms (most natably filter) that retains the sortedness of its input, and others (like transform) where if the I put is sorted by one comparator/projection it's typical for the output to be sorted by some other comparator (typically with a different projection).
Here are [slides to a talk I gave on CoreCpp 2023](https://docs.google.com/presentation/d/115iU6rA6gAUNErenmBIPqdl_HHlcYp9_YoYLOFSUNho/present) which discusses (among other things) potential enhancements to a ranges library - notably related to sorted ranges.

**Comment by roibarkan** [comment]
And btw, here's a [good academic talk](https://youtu.be/RjUpPr_Wzqo) about an efficient implementation of the 2-way merge algorithm for GPUs.
